day1/adventcode2015_day1_refactor.py

```diff
@@ -0,0 +1,32 @@
+with open('input.txt', 'r') as floors:
+    visited_floors: str = floors.readline().strip()
+
+
+def count_by_character(input_string, counter=0):
+    match input_string:
+        case '(':
+            counter += 1
+        case ')':
+            counter -= 1
+    return counter
+
+
+def count_santa_floor_and_basement_index(floors_input: str) -> tuple[int, int or None]:
+
+    brackets_counter: int = 0
+    basement_index: None = None
+
+    for floor_index, floor in enumerate(floors_input):
+
+        brackets_counter += count_by_character(floor)
+
+        if brackets_counter == -1 and basement_index is None:
```

Comment:
instead of hardcoding -1, I'd accept new 'target' parameter that defaults to -1, that way if anyone wants to pass let's say floor 2 and get on which index is that, it's possible. 

It gives more flexibility without changing current function invocation